src/theme-context-provider.tsx

```diff
@@ -91,6 +91,19 @@ const ThemeContextProvider: React.FC<Props> = ({theme, children, as, withoutStyl
     const isDarkModeEnabled = (colorScheme === 'auto' && isOsDarkModeEnabled) || colorScheme === 'dark';
     const colors: Colors = isDarkModeEnabled ? darkColors : lightColors;
 
+    const ref = React.useRef<HTMLDivElement>(null);
+    const [hasContentIsolation, setHasContentIsolation] = React.useState(false);
+
+    useIsomorphicLayoutEffect(() => {
+        // Set isolation: isolate to the parent of the provider. This way, we avoid content inside portals
+        // from being rendered under content that is inside the provider (in case it has z-index defined).
+        const root = ref.current?.parentElement;
+        if (root) {
+            root.style.isolation = 'isolate';
```

Comment:
I think that, when the ThemeContextProvider has the `as` prop, the `isolation: isolate` style should be applied to the create dom node, instead of applying it to the parent

Reply:
done